src/main.ts

```diff
@@ -16,8 +16,7 @@ export async function main() {
     const CONTAINER_NAME = `MICROSOFT_AZURE_CLI_${getCurrentTime()}_CONTAINER`;
     try {
         if (process.env.RUNNER_OS != 'Linux') {
-            core.setFailed('Please use Linux based OS as a runner.');

```

Comment:
Why doesn't `core.setFailed` cause failure?

Reply:
The description of `core.setFailed` is "Sets the action status to failed. When the action exits it will be with an exit code of 1.". 
However, in the previous code, it returned after setFailed.
https://github.com/Azure/cli/blob/1828f1caeefdc0631b30ab61f3c624b115f58c93/src/main.ts#L18-L20
This would cause `main()` didn't catch the error.
https://github.com/Azure/cli/blob/1828f1caeefdc0631b30ab61f3c624b115f58c93/src/main.ts#L77-L80
Then, `main()` exited with 0.
https://github.com/Azure/cli/blob/1828f1caeefdc0631b30ab61f3c624b115f58c93/src/entrypoint.ts#L4-L10
This is why `core.setFailed` doesn't cause failure in previous code.